comparison.py

```diff
@@ -0,0 +1,334 @@
+"""
+
+Author: Damian Pietroń
+Created: 2024-03-18
+Last modified: 2024-03-18
+
+Description: This script takes a path
+to a folder with the data and calculates
+the HRV parameters for the ABP and ECG signals.
+The calculated parameters are then merged into
+a single dataframe and saved as a csv file.
+The script also creates Bland-Altman plots for
+the calculated parameters.
+
+"""
+import neurokit2 as nk
+from pyhrv import frequency_domain as fd
+import matplotlib.pyplot as plt
+from pathlib import Path
+import pandas as pd
+import os
+import numpy as np
+import scipy.signal as sp_sig
+import scipy.fftpack as sp_fft  
+from tqdm import tqdm
+from datetime import datetime
+# The following lists contain the HRV parameters that will be calculated for the ABP and ECG signals.
+CONTENT = [
+    'HRV_SDNN',
+    'HRV_RMSSD',
+    'HRV_MeanNN',
+    'HRV_pNN50',
+    'HRV_pNN20',
+    'HRV_ApEn',
+    'HRV_SampEn',
+    'HRV_FuzzyEn',
+    'HRV_MSEn',
+    'HRV_ShanEn'
+]
+
+FREQ_CONTENT = [
+    'HRV_LF',
+    'HRV_HF',
+    'HRV_TP'
+]
+
+COLUMN_SETS_ECG = [
+    ['ekg[]', 'co2[co2]', 'etco2[mmHg]'], 
+    ['ecg', 'co2[mmHg]', 'etco2[mmHg]'],
+    ["ekg[ekg]", "co2[co2]", "etco2[etco2]"],
+    ['ekg[]', "co2[mm_Hg]", "etco2[mm_Hg]"]
+    ]
+
+COLUMN_SETS_ABP = [
+    ['abp_finger[abp_finger]', "co2[mmHg]", "etco2[mmHg]"], 
+    ['abp_cnap[mmHg]', "co2[mmHg]", "etco2[mmHg]"], 
+    ["abp_finger[abp_finger]", "co2[co2]", "etco2[etco2]"],
+    ["abp_finger[mm_Hg]", "co2[mm_Hg]", "etco2[mm_Hg]"]
+    ]
+
+def handle_path(path):
+    """
+    This function takes a path to a folder and returns a list of all the csv files in the folder.
+
+    Parameters
+    ----------
+    path : str
+        The path to the folder with the data.
+    
+    Returns
+    -------
+    csv_files : list
+        A list of all the csv files in the folder.
+    """
+    folder_path = Path(path)  # path to the folder with the data
+    csv_files = list(folder_path.glob("*.csv"))  # list of all the csv files in the folder
+    return csv_files, folder_path
+
+def get_columns_from_file(filename, column_sets, sep=',', decimal='.'):
+    """
+    This function reads a csv file and returns the columns that are specified in the column_sets list.
+
+    Parameters
+    ----------
+    filename : str
+        The name of the csv file.
+    column_sets : list
+        A list of lists with the column names that are searched for in the csv file.
+    sep : str, optional
+        The separator used in the csv file. By default ','.
+    decimal : str, optional
+        The decimal separator used in the csv file. By default '.'.
+    
+    Returns
+    -------
+    list
+        A list of the columns that are specified in the column_sets list.

```

Comment:
```python
"""
    list of pandas.Series or None
        Returns a list of pandas.Series corresponding to the columns specified in `column_sets` if all columns in any sublist are found, otherwise returns None.
"""
```